tinygrad/runtime/ops_clang.py

```diff
@@ -17,8 +17,9 @@ def __init__(self, name:str, prg:str):
     # TODO: is there a way to not write this to disk?
     fn = f"{tempfile.gettempdir()}/clang_{hashlib.md5(prg.encode('utf-8')).hexdigest()}.{args['ext']}"
     if not os.path.exists(fn):
-      subprocess.check_output(args=('clang -shared -O2 -Wall -Werror -x c '+args['cflags']+' - -o '+fn+'.tmp').split(), input=prg.encode('utf-8'))
-      os.rename(fn+'.tmp', fn)
+      _, tmp = tempfile.mkstemp()
```

Comment:
does this ever close the handle that `mkstemp` returns?

Reply:
That's a good point.

It seems there is no good way how to achieve what we need here in Python (generate a filename but not create a file): https://stackoverflow.com/questions/26541416/generate-temporary-file-names-without-creating-actual-file-in-python

We can either:
1. create a temporary directory (`tmp = os.path.join(tempfile.mkdtemp(), 'something')`), len clang create a file there, then remove both the file and the directory
2. use deprecated `mktemp`

What do you think @wozeparrot ?

Reply:
does it not work if we just add the current process pid to the filename?

Reply:
The PIDs can be recycled, so I don't think that's a good idea.

Reply:
it's only for the temporary file tho, and the pid is guaranteed to be unique as long as the process is alive which it will be while the tempfile is in use.

Reply:
even if same process calls it twice it will be sequential calls not parallel so the race condition won't happen

Reply:
yea, that would be nice.

Reply:
ack -> https://github.com/tinygrad/tinygrad/pull/1541